### ISSUE TITLE
match.params aren't URI decoded

### ISSUE THREAD

**Issue by gbenson-wikia** [ReactTraining/react-router#4605]
## Version
v4.0.0-beta.6

## Test Case
```
            <Route
                exact
                path="/example/:param_1"
                render={
                ({ match }) => {
                    console.log(match.params.param_1);
                    return (
                        <div />
                    );
                }
            }
            />
```

## Steps to reproduce
With the above route, visit `/example/this%20page%20has%20spaces%20here`

## Expected Behavior
`match.params.param_1` === `this page has spaces here`

## Actual Behavior
`match.params.param_1` === `this%20page%20has%20spaces%20here`

**Comment by pshrmn** [comment]
Semi-related to #4319.

**Comment by mjackson** [comment]
This is fixed (with a test) in 7279d22
